### PR TITLE
chore: use effective coalesing for selector cache keys and download button

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Fixes logical errors in coalescing in selector cache keys and download button. Not marked as a bug because these cases likely dont actually amount to any broken functionality (see comments below).

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

Very low risk.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

* Sanity check transaction history download `inputAddresses` and `outputAddresses` show `-` where empty rather than `"undefined"`
* Check portfolio balances around the app are unaffected by this change

## Screenshots (if applicable)

Tx history CSV working
image

### CODE REVIEW

File: src/pages/TransactionHistory/DownloadButton.tsx [hunk 1]
```diff
@@ -102,12 +102,12 @@ export const DownloadButton = ({ txIds }: { txIds: TxId[] }) => {
...
           : '-',
         inputCurrency: send?.asset?.symbol ?? send?.assetId ?? '-',
-        inputAddresses: `"${send?.from.join('\n')}"` ?? '-',
+        inputAddresses: send ? `"${send?.from.join('\n')}"` : '-',
         outputAmount: receive
           ? bnOrZero(fromBaseUnit(receive.value, receive.asset?.precision ?? 18)).toFixed()
           : '-',
         outputCurrency: receive?.asset?.symbol ?? receive?.assetId ?? '-',
-        outputAddresses: `"${receive?.from.join('\n')}"` ?? '-',
+        outputAddresses: receive ? `"${receive?.from.join('\n')}"` : '-',
```
Comment (woodenfurniture) [inline src/pages/TransactionHistory/DownloadButton.tsx:110]: if receive is undefined, we end up with `'"undefined"'`. 

`'"undefined"' ?? '-'` evaluates to `'"undefined"'` in all cases.

File: src/state/selectors.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ export const selectParamFromFilter = <T extends ParamFilterKey>(param: T) =>
     param => param,
   )(
     (_state: ReduxState, filter: Pick<ParamFilter, T> | null) =>
-      `${param}-${filter?.[param]}` ?? param,
+      `${param}-${filter?.[param] ?? param}`,
```
Comment (woodenfurniture) [inline src/state/selectors.ts:64]: if `filter` is undefined we get `'param-undefined'`. Coalescing did nothing here. But it's a valid string unique to the failure case so was not actually broken. The change here simply aligns the implementation with the intention.

Comment (kaladinlight) [reply src/state/selectors.ts:64]: feels like intention would be:
```
filter ? `${param}-${filter.[param]}` : param
```
so it is either `param` or `param-filter`?

Comment (kaladinlight) [reply src/state/selectors.ts:64]: I guess yours does match more closely the style in the selectors below now that I am looking at them

**Review by kaladinlight — APPROVED** [review]
lgtm!

**Pull request merged**